src/webgpu/api/operation/resource_init/buffer.spec.ts

```diff
@@ -309,3 +308,93 @@ remaining part of it will be initialized to 0.`
     const expectedBufferData = new Uint8Array(bufferSize);
     await t.CheckGPUBufferContent(dstBuffer, bufferUsage, expectedBufferData);
   });
+
+g.test('copy_texture_to_partial_buffer')
+  .desc(
+    `Verify when we copy from a GPUTexture into a GPUBuffer just after creating that GPUBuffer, the
+remaining part of it will be initialized to 0.`
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('bufferOffset', [0, 8, -16])
+      .combine('arrayLayerCount', [1, 3])
+      .combine('copyMipLevel', [0, 2])
+      .combine('rowsPerImage', [16, 20])
+      .filter(t => {
+        // We don't need to test the copies that will cover the whole GPUBuffer.
+        return !(t.bufferOffset === 0 && t.rowsPerImage === 16);
+      })
+  )
+  .fn(async t => {
+    const { bufferOffset, arrayLayerCount, copyMipLevel, rowsPerImage } = t.params;
+    const srcTextureFormat = 'r8unorm';
+    const textureSize = { width: 32, height: 16, depthOrArrayLayers: arrayLayerCount };
+
+    const srcTexture = t.device.createTexture({
+      format: srcTextureFormat,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+      size: textureSize,
+      mipLevelCount: copyMipLevel + 1,
+    });
+
+    const bytesPerRow = 256;
+    const layout = getTextureCopyLayout(
+      srcTextureFormat,
+      '2d',
+      [textureSize.width, textureSize.height, textureSize.depthOrArrayLayers],
+      {
+        mipLevel: copyMipLevel,
+        bytesPerRow,
+        rowsPerImage,
+      }
+    );
+
+    const extraBufferData = Math.abs(bufferOffset);
+    const bufferSize = layout.byteLength + extraBufferData;
+    const bufferUsage = GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST;
+    const dstBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: bufferUsage,
+    });
+
+    const encoder = t.device.createCommandEncoder();
+
+    // Initialize srcTexture
+    for (let layer = 0; layer < arrayLayerCount; ++layer) {
+      const renderPass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: srcTexture.createView({
+              baseArrayLayer: layer,
+              arrayLayerCount: 1,
+              baseMipLevel: copyMipLevel,
+            }),
+            loadValue: { r: 0.2 * (layer + 1), g: 0, b: 0, a: 0 },
```

Comment:
Won't this clamp to 1 as soon as the second layer? To have less float precision issues maybe we could use an integer texture format.